src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterModule.java

```diff
@@ -352,11 +355,13 @@ public CertificateStore getCertificateStore(MatchingServiceAdapterConfiguration
     }
 
     private Certificate cert(String keyName, String cert, Certificate.KeyUse keyUse) {
-        String certBody = cert
-                .replace("-----BEGIN CERTIFICATE-----", "")
-                .replace("-----END CERTIFICATE-----", "")
-                .replace(" ", "");
-        return new Certificate(keyName, certBody, keyUse);
+        try {
+            X509Certificate x509cert = X509Certificate.getInstance(cert.getBytes());
+            String certBody = Base64.encodeBase64String(x509cert.getEncoded());
+            return new Certificate(keyName, certBody, keyUse);
+        } catch (CertificateException e) {
+            throw new RuntimeException(e);
```

Comment:
If you can add a custom exception I'll add my approval to the PR - something like `InvalidCertificateException extends RuntimeException`

Reply:
I think @andy-paine has ideas about how this can use some common code we have elsewhere

Reply:
Nice.  This is an improvement that could be made before that happens though